pkg/scheduler/core/generic_scheduler.go

```diff
@@ -286,21 +282,22 @@ func (g *genericScheduler) Preempt(ctx context.Context, prof *profile.Profile, s
 		klog.Infof("%v potential nodes for preemption, first %v are: %v", len(potentialNodes), len(sample), sample)
 	}
 	var pdbs []*policy.PodDisruptionBudget
-	if g.pdbLister != nil {
-		pdbs, err = g.pdbLister.List(labels.Everything())
+	pdbLister := fh.SharedInformerFactory().Policy().V1beta1().PodDisruptionBudgets().Lister()
+	if pdbLister != nil {
+		pdbs, err = pdbLister.List(labels.Everything())
 		if err != nil {
 			return "", err
 		}
 	}
-	nodeNameToVictims, err := selectNodesForPreemption(ctx, prof, g.podNominator, state, pod, potentialNodes, pdbs)
```

Comment:
Can we remove podNominator now from generic scheduler?

Reply:
It's better to wait until all preemption functions get moved to preemption plugin, and then I can (1) revisit each function's signature in defaultpreemption package, and (2) genericScheduler should be easy by then to be cleaned up.